crates/sparrow-runtime/src/execute/operation/scan.rs

```diff
@@ -233,16 +222,38 @@ impl ScanOperation {
                 .change_context(Error::internal_msg("failed to create stream reader"))?
                 .map_err(|e| e.change_context(Error::internal_msg("failed to read batch")))
                 .boxed();
+
                 input_stream
             }
         };
 
+        // Streams until the stop signal has been received
```

Comment:
Does this mean that there are two usages here:
* Stop Signal is provided -> Streaming until stop signal
* Stop Signal is empty ->  Non-streaming scenario so read until input stream finishes

Do you mind updating the doc on this if so? This is the key piece that stops the stream so it's better to be over descriptive.